solution.rb

```diff
@@ -201,23 +204,63 @@
 ]
 
 most_expensive_products_by_store = []
-qw = []
+#loop in stores
 stores.each do |store|
-    price_a = []
-    hi_price = 0.0
-    
+    #for each store create these==
+    price_array = []
+    description_array = []
+    name_of_higer_product = ""
+    hiegher_price = 0.0
+    #loop inside each product inside stores
     store[:products].map do |product|
+        #gitting  prices and  names
         all_price = product[:price]
-        price_a.push(all_price.to_f)
+        prudct_name = product[:description]
+        #pushing price to price array
+        price_array.push(all_price.to_f)
+        #pushing product name to description array
+        description_array.push(prudct_name)
+    end # end of inner loop
+    #check what price is higher and make it the heigher price and get the product name as well
+    if price_array[0] > price_array[1]
+        hiegher_price = price_array[0]
+        name_of_higer_product = description_array[0]
+    else
+        hiegher_price = price_array[1]
+        name_of_higer_product = description_array[1]
     end
```

Comment:
This would only work if you only had to products, what happens if you have more? How can you fix that?
Would this work if jarir had these products:

```ruby
store_name: 'Jarir',
      products: [
          {
              description: 'Titanium',
              price: 9384.33
          },
          {
              description: 'Gold',
              price: 345.54
          }, 
           {
              description: 'Diamond',
              price: 1345.54
          }
      ]
```